optuna/_hypervolume/hssp.py

```diff
@@ -94,19 +92,18 @@ def _solve_hssp_on_unique_loss_vals(
 
     # The following logic can be used for non-unique rank_i_loss_vals as well.
     diff_of_loss_vals_and_ref_point = reference_point - rank_i_loss_vals
-    assert subset_size <= rank_i_indices.size
-    n_objectives = reference_point.size
+    (n_solutions, n_objectives) = rank_i_loss_vals.shape
+    n_solutions = int(n_solutions)  # MyPy Redefinition for NumPy v2.2.0.
     contribs = np.prod(diff_of_loss_vals_and_ref_point, axis=-1)
     selected_indices = np.zeros(subset_size, dtype=int)
     selected_vecs = np.empty((subset_size, n_objectives))
-    indices: np.ndarray[tuple[int, ...], np.dtype[Any]] = np.arange(
-        rank_i_loss_vals.shape[0], dtype=int
-    )
+    indices = np.arange(n_solutions)
     for k in range(subset_size):
         max_index = int(np.argmax(contribs))
         selected_indices[k] = indices[max_index]
         selected_vecs[k] = rank_i_loss_vals[max_index].copy()
-        keep = np.ones(contribs.size, dtype=bool)
+        assert n_solutions - k > 0
+        keep = np.ones(n_solutions - k, dtype=bool)
```

Comment:
Is it possible to use `contribs.size` as it is in the original logic?

Reply:
I tweaked the code and I made it.
But interestingly, the numpy sizing is now very sensitive to the cast information.

For example, if we replace `indices = np.arange(n_solutions)` with `indices = np.arange(int(n_solutions))`, `contribs.size` does not work anymore.

Anyways, your concern was resolved.